src/main/kotlin/com/jetbrains/snakecharm/codeInsight/completion/SMKRuleReferenceCompletionContributor.kt

```diff
@@ -0,0 +1,37 @@
+package com.jetbrains.snakecharm.codeInsight.completion
+
+import com.intellij.codeInsight.completion.*
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.ProcessingContext
+import com.jetbrains.snakecharm.lang.psi.SMKWorkflowLocalRulesStatement
+import com.jetbrains.snakecharm.lang.psi.SMKWorkflowRuleOrderStatement
+import com.jetbrains.snakecharm.lang.psi.SnakemakeFile
+
+class SMKRuleReferenceCompletionContributor : CompletionContributor() {
+    init {
+        extend(CompletionType.BASIC, RuleReferenceCompletionProvider.CAPTURE, RuleReferenceCompletionProvider)
+    }
+}
+
+object RuleReferenceCompletionProvider : CompletionProvider<CompletionParameters>() {
```

Comment:
PsiReference provide `getVariants()` method. Do we really need completion contributor here?